docs: added document on principles and agreements in the project

sholohov opened the pull request:
- the document for contributing has been moved to the root directory

sholohov commented on docs/principles-and-agreements.md:
```diff
@@ -0,0 +1,163 @@
+  - для работы с переменными окружения используется пакет `cross-env`
+  - для работы с архивами используется пакет `gzip-cli`
+- **идентификация компонента и коллизии имён** - для лучшего ориентирования не только в файловой структуре, но и при отладке кода в таких инструментах как: Browser DevTools и VueDevtools, рекомендуется:
+  - именовать vue файл в `kebab-case` пример `custom-component.vue`
```
> если для компонента помимо его самого надо вспомогашки:
> 
> ...
> component-name.json
> component-name.xxx
> ?

Я так думаю, не стоит мешать логику, данные и представление, vue-компонент, по сути - это представление и минимум логики, т.е. если есть папка components, там должны быть именно компоненты, данные, конфиги - другая ответственность, должны находится на уровне дата провайдера и конфигов соответственно.
Считаю нужно установить чёткие границы ответственности:
- `dataProvider`, `services` - отвечают за предоставление данных и методы для работы с ними, там же находятся моки.
- `components` - vue-компоненты, 1-файловые или 2-файловые, представление - шаблон+стили и js-логика
- `plugins` - кросс-проектные вспомогательные функции
- `config` - конфигурации, переменные для настройки приложения в зависимости от окружения
Это основные ответственности, про остальные - другой разговор.

sholohov replied:
Если в директории `component` должны лежать vue-компоненты, 1-файловые или 2-файловые, представление - шаблон+стили и js-логика, то есть у меня есть несколько вариантов:

sholohov replied:
Если в директории component должны лежать vue-компоненты, 1-файловые или 2-файловые, представление - шаблон+стили и js-логика, предлагаю несколько вариантов:
вариант 1:
```
comonents
    СomponentOne.vue // 1-файловый компонент
    СomponentTwo // 2-файловый компонент
        сomponentTwo.js
        сomponentTwo.sass
```
вариант 2:
```
comonents
    СomponentOne.vue // 1-файловый компонент
    СomponentTwo // 2-файловый компонент
        сomponent-two.js
        сomponent-two.sass
```
вариант 3:
```
comonents
    сomponent-one.vue // 1-файловый компонент
    сomponent-two // 2-файловый компонент
        сomponent-two.js
        сomponent-two.sass
```

я за 1, 2

euheny replied:
обсудим на дэйли

euheny commented on docs/principles-and-agreements.md:
```diff
@@ -0,0 +1,163 @@
+- **[Styleguide по Markdown](https://arcticicestudio.github.io/styleguide-markdown/)** -
+рекомендации по ведению документации
+- **именование методов и функций**
+  происходит по схеме `function [action][triger][event] () { ... }`, `event` - может не быть:
```
обсудим на дэйли

euheny commented on docs/principles-and-agreements.md:
```diff
@@ -0,0 +1,163 @@
+
+## Соглашения
+
+- **[Styleguide по Markdown](https://arcticicestudio.github.io/styleguide-markdown/)** -
```
1 - маркдаун не настолько адово гибкий, чтобы упарываться на его гайды
2 - я бы разделял файлы
3 - как предложение (не только к этой доке) можно взять airbnb как основу гайдов и по возможности тащить данные отуда (касательно всех зон) тем более что и линтер у нас оттуда частично

sholohov commented:
changelog: updated agreement points

- removed agreement for .md documentation
- updated agreement for event handlers
- updated agreement for component identity and names collision

euheny commented on docs/principles-and-agreements.md:
````diff
@@ -0,0 +1,152 @@
+  - для работы с архивами используется пакет `gzip-cli`
+- **идентификация компонента и коллизии имён** - для лучшего ориентирования в файловой структуре проекта и во избежание коллизии имён и возможности лёгкого масштабирования проекта рекомендуется использовать паттерн, где дочерний компонент в имени содержит префиксом имя родителя, пример:
+  ```
+  automation
````
думаю этот блок стоит выпилить и перенести в раздел нэйминга

sholohov replied:
Ок. Но что б не сочинять дважды, может закинь куда-нибудь себе в доку, прежде чем я уберу отсюда

euheny replied:
может замутить вайл типа _temp-agregate.md и туда все пихать, а же в конечных файлах оставлять только тематическое?

sholohov replied:
Поддерживаю. Переименую файл. Может как-то в одной ветке на доками работать?

euheny replied:
пока из своей все впили. потм возможно и в одной. я сейчас смотрю варианты нормльного ui для этого дела, так что скоро совсем определимся как оно все будет

euheny reviewed (changes requested):
в дальнейшем надо будет части этой доки или перегреппировать или перенести, чтобы оталась информация только касательная srp

elena-vlasova commented on docs/principles-and-agreements.md:
```diff
@@ -0,0 +1,152 @@
+  │
+  ├───rule
+  │   │   AutomationRuleSetup.vue
+  │   │   AutomationRuleSetup-item.vue
```
Почему item через дефис написано?

sholohov replied:
Спасибо. Не заметил, когда из кебеба конвертировал

euheny commented:
ты вообще получается примеры выпилил из доки по ответственности?

sholohov commented:
> ты вообще получается примеры выпилил из доки по ответственности?

Да. Мы ж вроде уже решили, что размещаем только соглашение о том что мы юзаем SRP и даём ссыль на доку гугла, если тема интересна пусть сам гуглит, что нет смысла давать кучу примеров, если по этому есть масса статей. Пример был в соглашениях "один метод - одно действие (группировка действий)" это что-то около темы и только про методы, этого мало, после того, как вынес SRP и Декомпозицию в отдельную доку, удалил.

sholohov commented:
Добавил пример для SRP
